R/summariseClinicalRecords.R

```diff
@@ -47,80 +47,77 @@
 
-  assertLogical(inObservation, length = 1)
-  assertLogical(standardConcept, length = 1)
-  assertLogical(sourceVocabulary, length = 1)
-  assertLogical(domainId, length = 1)
-  assertLogical(typeConcept, length = 1)
-
-  resultList <- list()
-  for(tab in tableNames) {
-    cli::cli_progress_bar()
-    resultList[[tab]] <- summariseClinicalRecord(
-      cdm = cdm,
-      tableName = tab,
-      recordsPerPerson = recordsPerPerson,
-      inObservation = inObservation,
-      standardConcept = standardConcept,
-      sourceVocabulary = sourceVocabulary,
-      domainId = domainId,
-      typeConcept = typeConcept
-    )
-  }
-
-  result <- dplyr::bind_rows(resultList)
+  omopgenerics::assertLogical(inObservation, length = 1)
+  omopgenerics::assertLogical(standardConcept, length = 1)
+  omopgenerics::assertLogical(sourceVocabulary, length = 1)
+  omopgenerics::assertLogical(domainId, length = 1)
+  omopgenerics::assertLogical(typeConcept, length = 1)
+
+  result <- purrr::map(omopTableName,
+                       function(x) {
+                         if(cdm[[x]] |> dplyr::tally() |> dplyr::pull("n") == 0) {
+                           cli::cli_warn(paste0(x, " omop table is empty. Returning an empty summarised omop table."))
+                           return(omopgenerics::emptySummarisedResult())
+                         }
+                         summariseClinicalRecord(x,
+                                                 cdm = cdm,
+                                                 recordsPerPerson = recordsPerPerson,
+                                                 inObservation = inObservation,
+                                                 standardConcept = standardConcept,
+                                                 sourceVocabulary = sourceVocabulary,
+                                                 domainId = domainId,
+                                                 typeConcept = typeConcept)
+                       }
+                       ) |>
+    dplyr::bind_rows()
 
   return(result)
 }
 
 #' @noRd
-summariseClinicalRecord <- function(cdm,
-                                    tableName,
-                                    recordsPerPerson = c("mean", "sd", "median", "q25", "q75", "min", "max"),
-                                    inObservation = TRUE,
-                                    standardConcept = TRUE,
-                                    sourceVocabulary = FALSE,
-                                    domainId = TRUE,
-                                    typeConcept = TRUE) {
-
-  assertClass(cdm[[tableName]], "omop_table")
-  omopTable <- cdm[[tableName]] |> dplyr::ungroup()
-
-  if ("observation_period" == tableName) {
+summariseClinicalRecord <- function(omopTableName, cdm, recordsPerPerson,
+                                    inObservation, standardConcept,
+                                    sourceVocabulary, domainId, typeConcept,
+                                    call = parent.frame(3)) {
```

Comment:
3?

Reply:
So that the parent frame refers to the main function summariseClinicalRecords() and not the purrr::map() call